crates/topos-tce-transport/src/lib.rs

```diff
@@ -101,14 +112,61 @@ pub enum ProtocolEvents {
     /// Indicates that 'echo' message broadcasting is required
     Echo {
         certificate_id: CertificateId,
+        signature: Signature,
+        validator_id: ValidatorId,
     },
     /// Indicates that 'ready' message broadcasting is required
     Ready {
         certificate_id: CertificateId,
+        signature: Signature,
+        validator_id: ValidatorId,
     },
     /// For simulation purpose, for now only caused by ill-formed sampling
     Die,
 
     /// Stable Sample
     StableSample,
 }
+
+#[derive(Debug, Error)]
+pub enum ValidatorIdConversionError {
+    #[error("Failed to parse address string as H160")]
+    ParseError,
+    #[error("Failed to convert byte array into H160")]
+    InvalidByteLength,
+}
+
+#[derive(Clone, Copy, Debug, Serialize, Deserialize, Eq, PartialEq, Hash)]
+pub struct ValidatorId(H160);
```

Comment:
I'm wondering if the `ValidatorId` should be part of `tce-transport` or `core`. Let's see that in a next PR